### PR TITLE
Bump AWSSDK.Core dependency version

### PR DESCRIPTION
**Pull request by lupengamzn**
*Issue #, if available:*

#147 #153 

*Description of changes:*

Bumped `AWSSDK.Core` dependency version to **3.5.1.23**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sdk/src/Core/AWSXRayRecorder.Core.csproj [hunk 1]
```diff
@@ -38,8 +38,8 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="AWSSDK.Core" Version="3.3.25.1" />
-    <PackageReference Include="AWSSDK.XRay" Version="3.3.3" />
+    <PackageReference Include="AWSSDK.Core" Version="3.5.1.23" />
```
Comment (srprash) [inline sdk/src/Core/AWSXRayRecorder.Core.csproj:41]: As discussed, can we provide a range of version here as `Version="[3.3.25.1,4.0)`

File: sdk/src/Core/AWSXRayRecorder.Core.csproj [hunk 2]
```diff
@@ -38,8 +38,8 @@
...
+    <PackageReference Include="AWSSDK.XRay" Version="3.5.2.16" />
```
Comment (srprash) [inline sdk/src/Core/AWSXRayRecorder.Core.csproj:42]: Same here. Provide a range.

File: sdk/src/Handlers/AspNet/AWSXRayRecorder.Handlers.AspNet.csproj [hunk 3]
```diff
@@ -30,7 +30,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="AWSSDK.Core" Version="3.3.25.1" />
+    <PackageReference Include="AWSSDK.Core" Version="3.5.1.23" />
```
Comment (srprash) [inline sdk/src/Handlers/AspNet/AWSXRayRecorder.Handlers.AspNet.csproj:33]: Do we need to depend on AWSSDK.Core for this project? If not, can we remove it?

File: sdk/src/Handlers/AwsSdk/AWSXRayRecorder.Handlers.AwsSdk.csproj [hunk 4]
```diff
@@ -42,7 +42,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="AWSSDK.Core" Version="3.3.25.1" />
+    <PackageReference Include="AWSSDK.Core" Version="3.5.1.23" />
```
Comment (srprash) [inline sdk/src/Handlers/AwsSdk/AWSXRayRecorder.Handlers.AwsSdk.csproj:45]: Provide a range from 3.3.25.1 to 4.0

File: sdk/src/Handlers/SqlServer/AWSXRayRecorder.Handlers.SqlServer.csproj [hunk 5]
```diff
@@ -30,7 +30,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="AWSSDK.Core" Version="3.3.25.1" />
+    <PackageReference Include="AWSSDK.Core" Version="3.5.1.23" />
```
Comment (srprash) [inline sdk/src/Handlers/SqlServer/AWSXRayRecorder.Handlers.SqlServer.csproj:33]: This also is probably not required

File: sdk/src/Handlers/System.Net/AWSXRayRecorder.Handlers.System.Net.csproj [hunk 6]
```diff
@@ -30,7 +30,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="AWSSDK.Core" Version="3.3.25.1" />
+    <PackageReference Include="AWSSDK.Core" Version="3.5.1.23" />
```
Comment (srprash) [inline sdk/src/Handlers/System.Net/AWSXRayRecorder.Handlers.System.Net.csproj:33]: Same here. Probably we can remove this dependency

**Review by srprash — CHANGES_REQUESTED** [review]

File: sdk/src/Core/AWSXRayRecorder.Core.csproj (hunk 1 above)
Comment (lupengamzn) [reply sdk/src/Core/AWSXRayRecorder.Core.csproj]: Updated

File: sdk/src/Handlers/AspNet/AWSXRayRecorder.Handlers.AspNet.csproj (hunk 3 above)
Comment (lupengamzn) [reply sdk/src/Handlers/AspNet/AWSXRayRecorder.Handlers.AspNet.csproj]: Yes, we need the `Logger` from `AWSSDK.Core` at this moment

File: sdk/src/Handlers/AwsSdk/AWSXRayRecorder.Handlers.AwsSdk.csproj (hunk 4 above)
Comment (lupengamzn) [reply sdk/src/Handlers/AwsSdk/AWSXRayRecorder.Handlers.AwsSdk.csproj]: Updated

File: sdk/src/Handlers/SqlServer/AWSXRayRecorder.Handlers.SqlServer.csproj (hunk 5 above)
Comment (lupengamzn) [reply sdk/src/Handlers/SqlServer/AWSXRayRecorder.Handlers.SqlServer.csproj]: We need `Logger` from `AWSSDK.Core`

File: sdk/src/Handlers/System.Net/AWSXRayRecorder.Handlers.System.Net.csproj (hunk 6 above)
Comment (lupengamzn) [reply sdk/src/Handlers/System.Net/AWSXRayRecorder.Handlers.System.Net.csproj]: Same as above

File: sdk/src/Core/AWSXRayRecorder.Core.csproj (hunk 2 above)
Comment (lupengamzn) [reply sdk/src/Core/AWSXRayRecorder.Core.csproj]: Updated

File: sdk/src/Handlers/AspNet/AWSXRayRecorder.Handlers.AspNet.csproj (hunk 3 above)
Comment (srprash) [reply sdk/src/Handlers/AspNet/AWSXRayRecorder.Handlers.AspNet.csproj]: From what I understand, if the AspNet project references the `AWSXRayRecorder.Core` project which uses AWSSDK.Core, this project should be able to use the `Logger` from AWSSDK.Core as transitive dependency?

Comment (lupengamzn) [reply sdk/src/Handlers/AspNet/AWSXRayRecorder.Handlers.AspNet.csproj]: Nice point, updated

**Review by srprash — APPROVED** [review]
LGTM

**Pull request merged**